### PR TITLE
fix: run build prior to release

### LINKED ISSUE #162: 🐛 Bug: published npm packages have src/ but not lib/
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

npm packages should have compiled outputs. `.js` files and such.

### Actual

Looks like they don't 😢 

### Additional Info

Examples:

* https://www.npmjs.com/package/remark-cloudinary-docusaurus/v/0.1.1?activeTab=explore
* https://www.npmjs.com/package/template-typescript-node-package/v/1.27.6?activeTab=explore

### PR DESCRIPTION
**Pull request by johnnyreilly**
## PR Checklist

- [x] Addresses an existing open issue: fixes #162
- [x] That issue was marked as [accepting prs](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This PR adds in a build step to the `release.yml` so the build artefacts exist for publishing. It also excludes `src` from publishing as it bloats package size and isn't used.

PR requested here: https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues/147#issuecomment-1364555919

It also ensure vitest doesn't try and run tests in the `lib` folder.

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ coverage/
 cspell.json
 pnpm-lock.yaml
 tsconfig*.json
+src
```
Comment (JoshuaKGoldberg) [inline .npmignore:11]: Nit: don't mind me, just alphabetizing.

```suggestion
src
tsconfig*.json
```

**Review by JoshuaKGoldberg — APPROVED** [review]
:100: Perfect, thanks @johnnyreilly!

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @johnnyreilly for code

**Comment by johnnyreilly** [comment]
Pleasure!

**Pull request merged**